### ISSUE TITLE
Reading Cigar for reads in reverse orientation

### ISSUE THREAD

**Issue by erondeau** [CNuge/snp_genome_placement#2]
Hey Cam,

I love the script - simple concept and easy to use! 

Not sure if you are still working with it, but I noticed a small issue. I'm not overly familiar with python, so I came up with a solution in R, but I suspect it is an easy fix. 

In `adjust_bp`, the cigar_dat is being read from left to right, and once the position of the SNP is reached it stops. But for reads aligned in reverse complement, the `bp_of_snp` is provided with respect to the original read orientation. Therefore, I believe the cigar string needs to be read from right to left. I noticed this where I had a cigar of 113M23S and flag == 16. My SNP position of 36 wasn't being changed, even though there was soft-clipping of 23 on what had been the first few bases of the read in the original orientation. 

I wanted to understand the issue, so I wrote it in R which I'm more familiar with: 

```adjust_bp <- function(bp_of_snp, cigar_dat, flag){
  change_to_bp <- 0
  bp_scan <- 0
  if (flag == 16 || flag == 272){
      cigar_dat_2 <- rev(cigar_dat)
  } else {
      cigar_dat_2 <- cigar_dat
  }
  for (j in 1:length(cigar_dat_2)){
...
```

I think this is what is needed - reverse the cigar data if flag == 16, then scan through until you reach the SNP - or read the cigar_dat from right to left if flag == 16. I tried to come up with a similar solution in python, but I think it is just a bit out of my understanding. 

Thanks,

Eric

**Comment by CNuge** [comment]
Hi Eric,

Thanks for pointing this out, your detailed notes made finding the source of the issue a breeze. I made the bug fix by adding the flag based logic to the `adjust_bp` function as you indicated. If you're curious here is what that looks like in python (if flag is one of the two values, then reverse the list and proceed as usual):
```
	if flag == 16 or flag == 272:
		cigar_dat = cigar_dat[::-1]		
```
I also made the necessary changes to the functions that call  `adjust_bp`  so they pass the flag data from the main file down to the adjustment function. A unit test with your specific example was also added.

Thanks for pointing it out, I hope the code has been of some use and that this resolves the issue!
-Cam
